src/main/java/org/openlmis/buq/domain/sourceoffund/SourceOfFund.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * This program is part of the OpenLMIS logistics management information system platform software.
+ * Copyright © 2017 VillageReach
+ *
+ * This program is free software: you can redistribute it and/or modify it under the terms
+ * of the GNU Affero General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details. You should have received a copy of
+ * the GNU Affero General Public License along with this program. If not, see
+ * http://www.gnu.org/licenses.  For additional information contact info@OpenLMIS.org.
+ */
+
+package org.openlmis.buq.domain.sourceoffund;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Table;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import lombok.ToString;
+import org.javers.core.metamodel.annotation.TypeName;
+import org.openlmis.buq.domain.BaseEntity;
+
+@Entity
+@TypeName("SourceOfFund")
+@Table(name = "sources_of_funds", schema = "buq")
+@NoArgsConstructor
+@EqualsAndHashCode(callSuper = true)
+@ToString(callSuper = true)
+public class SourceOfFund extends BaseEntity {
+
+  @Column(nullable = false, unique = true)
```

Comment:
I think you should use `@NotBlank` annotation here also.
The `nullable = false` only adds (if you use the `ddl-auto`) a not-null constraint to this column for this table schema.
With that said - having `nullable = false` you don't perform any validation before calling the DB.
With `@NotBlank` annotation, you wouldn't make the call to the DB if the `name` would be null/empty string

Reply:
I changed the annotations to @NotBlank